core/src/test/scala/com/salesforce/op/filters/PreparedFeaturesTest.scala

```diff
@@ -157,6 +162,68 @@ class PreparedFeaturesTest extends FlatSpec with TestSparkContext {
     testCorrMatrix(allResponseKeys2, CorrelationType.Spearman, expected)
   }
 
+  it should "correctly transform date features when time period is specified" in {
```

Comment:
this test has to be split into separate cases or be a for loop not a copy/paste

Reply:
done, there's a single function for running each check.